File: ObjCVerify/ObjCVerify/Examples/Controller/Tabbar/Right/WYRightController.m
//
//  WYRightController.m
//  ObjCVerify
//
//  Created by guanren on 2025/10/1.
//

#import "WYRightController.h"
#import <WYBasisKitObjC/WYBasisKitObjC.h>
#import "AppEventDelegate.h"

@interface WYRightController ()<AppEventDelegate>

@end

@implementation WYRightController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = [UIColor whiteColor];
    
    [WYEventHandler registerWithEvent:AppEventButtonDidMove target:self handler:^(id _Nullable data) {
        WYLog(@"data = %@, controller = %@",data, NSStringFromClass([self class]));
    }];
    
    [WYEventHandler registerWithEvent:AppEventButtonDidReturn target:self handler:^(id _Nullable data) {
        WYLog(@"data = %@, controller = %@",data, NSStringFromClass([self class]));
    }];
    
    __weak typeof(self) weakSelf = self;
    [WYEventHandler registerWithEvent:AppEventDidShowBannerView target:self handler:^(id _Nullable data) {
        __strong typeof(weakSelf) strongSelf = weakSelf;
        [strongSelf didShowBannerViewWithData: data];
    }];
}

- (void)didShowBannerViewWithData:(NSString *)data {
    WYLog(@"data = %@, controller = %@",data, NSStringFromClass([self class]));
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: SwiftVerify/SwiftVerify/Examples/Controller/Tabbar/right/WYRightController.swift
//
//  WYRightController.swift
//  WYBasisKit
//
//  Created by 官人 on 2020/12/3.
//  Copyright © 2020 官人. All rights reserved.
//

import UIKit

class WYRightController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        
        view.backgroundColor = .white
        
        WYEventHandler.shared.register(event: AppEvent.buttonDidMove, target: self) { data in
            if let stringValue = data {
                WYLogManager.output("data = \(stringValue), controller: \(type(of: self))")
            }
        }
        
        WYEventHandler.shared.register(event: AppEvent.buttonDidReturn, target: self) { data in
            if let stringValue = data {
                WYLogManager.output("data = \(stringValue), controller: \(type(of: self))")
            }
        }
        
        WYEventHandler.shared.register(event: AppEvent.didShowBannerView, target: self) { [weak self] data in
            if let dataString = data as? String,
               let delegate = self {
                delegate.didShowBannerView(data: dataString)
            }
        }
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}

extension WYRightController: AppEventDelegate {
    
    func didShowBannerView(data: String) {
        WYLogManager.output("data = \(data), controller: \(type(of: self))")
    }
}
